### ISSUE TITLE
How to solve 'nan' loss in training?

### ISSUE THREAD

**Issue by Chen1399** [fjiang9/NKF-AEC#4]
'w' will diverge during training, resulting in an INF situation. The loss will be 'nan' or a very big number. How to solve it? I try to add a 'Tanh' after W, which is useful.

**Comment by fjiang9** [comment]
You can use shorter audio segments at the beginning of training. Other 'warm up' training skills may also be good solutions to the 'nan' issue.

**Comment by Chen1399** [comment]
Thanks. I solved the problem.

**Comment by fjiang9** [comment]
Glad to hear that. Thanks for your interest!

**Comment by meadow163** [comment]
> Thanks. I solved the problem.

could you tell me how you solve this issue please?

**Comment by Chen1399** [comment]
> 
Adjust learning rate and training file length. And the init weight is important.

**Comment by TeaPoly** [comment]
> Adjust learning rate and training file length. And the init weight is important.

Is there any recommended way to initialize the weights?

**Comment by Chen1399** [comment]
> Is there any recommended way to initialize the weights?

Initialize the weights by the way of 'autodsp'. And use the very short wav to train is also the way.

**Comment by TeaPoly** [comment]
> > Is there any recommended way to initialize the weights?
> 
> Initialize the weights by the way of 'autodsp'. And use the very short wav to train is also the way.

Thanks for you reply.

**Comment by shenbuguanni** [comment]
I find the echo_hat is very easy clip, then loss will be nan ,  "very short wav " is mean 1s? or other?

**Comment by shenbuguanni** [comment]
please ask the ‘self.kg_net.init_hidden’ is need to use in trainning stage?  what about you say 'Initialize the weights', is mean GRU or Dense?

**Comment by Chen1399** [comment]
> I find the echo_hat is very easy clip, then loss will be nan , "very short wav " is mean 1s? or other?

Maybe shorter, which made the loss isn't nan.

**Comment by Chen1399** [comment]
> please ask the ‘self.kg_net.init_hidden’ is need to use in trainning stage? what about you say 'Initialize the weights', is mean GRU or Dense?
‘self.kg_net.init_hidden' is the state of gru. Using the shorter wav to train is a way to have a greater init-weights.

**Comment by Chen1399** [comment]
> please ask the ‘self.kg_net.init_hidden’ is need to use in trainning stage? what about you say 'Initialize the weights', is mean GRU or Dense?

‘self.kg_net.init_hidden' is the state of gru. Using the shorter wav to train is a way to have a greater init-weights.

**Comment by shenbuguanni** [comment]
I don't understand why shorter (maybe 0.5s) audio can avoid Nan，what is your batch_size?

**Comment by BruceWeiii** [comment]
> 谢谢。我解决了这个问题。

Hello, have you reproduced the results of the paper? Could you describe your training process? I modified the test code for training and found that the results were poor. Thank u!
